packages/test-base/src/androidAndroidTest/kotlin/io/realm/kotlin/test/shared/CompactOnLaunchTests.kt

```diff
@@ -65,14 +64,59 @@ class CompactOnLaunchTests {
         assertEquals(Realm.DEFAULT_COMPACT_ON_LAUNCH_CALLBACK, config.compactOnLaunchCallback)
     }
 
+    private val Int.MB get() = (this * 1024 * 1024).toLong()
+
+    private val Int.Bytes get() = (this).toLong()
+
     @Test
     fun defaultCallback_boundaries() {
+        // This callback will only return [True] if the file is above 50 MB and 50% or more of the space can be reclaimed.
         val callback = Realm.DEFAULT_COMPACT_ON_LAUNCH_CALLBACK
-        assertFalse(callback.shouldCompact(50 * 1024 * 1024, 40 * 1024 * 1024))
-        assertFalse(callback.shouldCompact(50 * 1024 * 1024 + 8, 25 * 1024 * 1024))
-        assertFalse(callback.shouldCompact(50 * 1024 * 1024 + 8, 25 * 1024 * 1024 + 3))
-        assertTrue(callback.shouldCompact(50 * 1024 * 1024 + 8, 25 * 1024 * 1024 + 4))
-        assertTrue(callback.shouldCompact(50 * 1024 * 1024 + 8, 25 * 1024 * 1024 + 5))
+
+        fun assert(shouldCompact: Boolean, fileSize: Long, usage: Double) {
+            assertEquals(
+                expected = shouldCompact,
+                actual = callback.shouldCompact(
+                    totalBytes = fileSize,
+                    usedBytes = (fileSize * usage).toLong()
+                ),
+                message = "fileSize: $fileSize usage: $usage"
+            )
+        }
+
+        // File size <= 50MB, it will never reclaim
+        assert(
```

Comment:
I see the motivation for doing wrappers like this, but I tend to find it less clear to follow the assertion. Don't know the right balance though 😞

Reply:
I kinda this way of writing the assertions, it makes it more easy to see what is being tested. (if you trust that the assertion code works :D)

Reply:
I think that is more expressive as it uses the concepts of what we are testing, like file usage as a percentage. But it adds another layer of complexity to the test...

In my opinion, it should be ok writing our own asserters if we keep them simple and they facilitate writing and understanding tests. For example, there are asserters for comparing collection contents instead of comparing them manually.

But yes, we should also add some tests for any new asserter 😅

Reply:
It is a great improvement to the old tests. But the improvement lies more or less in the helper `MB` and `B` converters whereas hiding the assertions makes it more tricky to reason about what is actually asserted. I guess it also is just down to the naming. `assert` sounds like it is comparing two values, but it is actually doing all the invocations under the hood. `testShouldCompact` would be more appropriate.

But you could also just do
```
DEFAULT_COMPATCH_ON_LAUNCH_CALLBACK.let {
   assertFalse(shouldCompact(50.MB, 50.MB * 0.5))
   ...
}
```
If the `fileSize` and `usage` named parameters really brings so much clarity, then we should consider adding it to the callback methods in a future API.